src/toil/job.py

```diff
@@ -41,55 +42,54 @@
 logger = logging.getLogger( __name__ )
 
 
-class Job(object):
+class ResourceRequirementMixin(object):
     """
-    Class represents a unit of work in toil.
+    Inherit from this class to add resource requirements for an object
     """
-    def __init__(self, memory=None, cores=None, disk=None, preemptable=None, checkpoint=False):
+    def __init__(self, memory=None, cores=None, disk=None):
+        self._cores = self._parseResource('cores', cores)
+        self._memory = self._parseResource('memory', memory)
+        self._disk = self._parseResource('disk', disk)
+        self._config = None
+
+    @property
+    def disk(self):
+        """
+        This property insures that even if the user does not pass a resource requirment, the
```

Comment:
Also no need to duplicate that # comment in every method. Maybe it should go above the first property.